base/base/find_symbols.h

```diff
@@ -34,10 +35,52 @@
   * If no such characters, returns nullptr.
   */
 
+struct SearchSymbols
+{
+    static constexpr auto BUFFER_SIZE = 16;
+
+    SearchSymbols() = default;
+
+    explicit SearchSymbols(std::string in)
+        : str(std::move(in))
+    {
+#if defined(__SSE4_2__)
+        if (str.size() > BUFFER_SIZE)
+        {
+            throw std::runtime_error("SearchSymbols can contain at most " + std::to_string(BUFFER_SIZE) + " symbols and " + std::to_string(str.size()) + " was provided\n");
```

Comment:
I believe `DB::Exception` should not be thrown here since it's in the `base` directory. Should it be a `std::exception` or a custom exception?

Reply:
Are you sure about that? I do not see any `DB::Exception` being thrown in this directory, only custom defined ones. I also see the below comment on `base/throwError.h`. 

`/// DB::Exception is not defined yet. It'd better to throw Poco::Exception but we do not want to include any big header here, even <string>.`

Reply:
Is this code expected to be unreachable? If yes, `std::runtime_error` is fine

Reply:
SSE42 expects needle / search symbol buffer to contain 16 bytes. Hence, the exception. I would say it's reachable.